internal/controller/flow_falco.go

```diff
@@ -0,0 +1,160 @@
+func parsePodNetworkInfo(input string) (FalcoEvent, error) {
+	var info FalcoEvent
+
+	// Regular expression to extract the key-value pairs from the input string
+	re := regexp.MustCompile(`\b(\w+)=([^\s)]+)`)
+	matches := re.FindAllStringSubmatch(input, -1)
+
+	for _, match := range matches {
+		if len(match) == 3 {
+			key, value := match[1], match[2]
+			switch key {
+			case "srcip":
+				info.SrcIP = value
+			case "dstip":
+				info.DstIP = value
+			case "srcport":
+				info.SrcPort = value
+			case "dstport":
+				info.DstPort = value
+			case "proto":
+				info.Proto = value
+			case "ipversion":
+				info.IpVersion = value
+			}
+		}
+	}
+	return info, nil
+}
+
+// NewFalcoEventHandler creates a new HTTP handler function for processing Falco events.
+func NewFalcoEventHandler(eventChan chan<- string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var body struct {
+			Output string `json:"output"`
+		}
+
+		if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		eventChan <- body.Output
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+// filterIllumioTraffic filters out events related to Illumio network traffic.
+func filterIllumioTraffic(body string) bool {
+	return strings.Contains(body, "illumio_network_traffic")
+}
+
+func convertFalcoEventToFlow(event FalcoEvent) (*pb.FalcoFlow, error) {
+	layer3Message, err := createLayer3Message(event.SrcIP, event.SrcIP, event.IpVersion)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Layer3 message falco flows: %v", err)
+	}
+
+	srcPort, err := strconv.ParseUint(event.SrcPort, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("invalid source port: %v", err)
+	}
+
+	dstPort, err := strconv.ParseUint(event.DstPort, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("invalid destination port: %v", err)
+	}
+
+	layer4Message, err := CreateLayer4Message(event.Proto, uint32(srcPort), uint32(dstPort), event.IpVersion)
+	if err != nil {
+		return nil, fmt.Errorf("could not create Layer4 Message for Falco flow %v", err)
+	}
+
+	flow := &pb.FalcoFlow{
+		Layer3: layer3Message,
+		Layer4: layer4Message,
+	}
+
+	return flow, nil
+}
+
+func createLayer3Message(source string, destination string, ipVersion string) (*pb.IP, error) {
+	if ipVersion == "ipv4" {
+		return &pb.IP{Source: source, Destination: destination, IpVersion: pb.IPVersion_IP_VERSION_IPV4}, nil
+	} else if ipVersion == "ipv6" {
+		return &pb.IP{Source: source, Destination: destination, IpVersion: pb.IPVersion_IP_VERSION_IPV6}, nil
+	}
+	return &pb.IP{Source: source, Destination: destination, IpVersion: pb.IPVersion_IP_VERSION_IP_NOT_USED_UNSPECIFIED}, nil
+}
+
+// CreateLayer4Message converts event protocol and ports to a Layer4 proto message
+func CreateLayer4Message(proto string, srcPort, dstPort uint32, ipVersion string) (*pb.Layer4, error) {
+	switch proto {
```

Comment:
Have you confirmed that Falco supports ICMP?
https://github.com/falcosecurity/libs/blob/230ddfb91d075241391697900ab4281aa67fe108/userspace/libsinsp/utils.cpp#L1026-L1052

Reply:
Seems like their docs may be different than implementation you linked.... https://falco.org/docs/reference/rules/supported-fields/ under fd.l4proto it states icmp is supported type.